lang/src/main/java/net/openhft/lang/io/Bits.java

```diff
@@ -0,0 +1,272 @@
+package net.openhft.lang.io;
+
+import java.lang.reflect.Field;
+import java.nio.ByteOrder;
+
+import org.jetbrains.annotations.NotNull;
+
+import sun.misc.Unsafe;
+
+public class Bits {
+
+  @NotNull
+  @SuppressWarnings("ALL")
+  public static final Unsafe UNSAFE;
+
+  public static final String MEMORY_ALIGNED_ACCESS_PROPERTY = "net.openhft.io.memory.aligned.access";
+  public static final String BYTE_ORDER_BIGENDIAN_PROPERTY = "net.openhft.io.byte.order.bigEndian";
+
+  public static final ByteOrder NATIVE_BYTE_ORDER = ByteOrder.nativeOrder();
+
+  /** Unaligned flag. */
+  protected static final boolean ALIGNED = aligned();
+
+  /** Big endian. */
+  protected static final boolean BIG_ENDIAN = bigEndian();
+
+  static {
+    try {
+      @SuppressWarnings("ALL")
+      Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
+      theUnsafe.setAccessible(true);
+      UNSAFE = (Unsafe) theUnsafe.get(null);
+    } catch (Exception e) {
+      throw new AssertionError(e);
+    }
+  }
+
+  public static short getShort(long addr) {
+    return ALIGNED ? getShortByByte(addr, BIG_ENDIAN) : UNSAFE.getShort(addr);
+  }
+
+  public static void putShort(long addr, short v) {
+    if (ALIGNED)

```

Comment:
Given the if argument is static, can you use static init-level dispatch instead of IFs in every method please? We don't want to add overhead of checking a condition when it can be avoided.

Reply:
If the value is `static final` the JIT compiler should perform dead code elimination on the branch that will never be executed. You could test with a JMH benchmark and check the assembly output.

Reply:
Eliminating the IFs would be possible with a separate BitsAligned and BitsUnaligned but that still leaves the BIG-ENDIAN checks. Although those could also be eliminated with the same approach I am not sure the code would be cleaner then as it would need 4 separate classes to handle all the combinations.